fs.go

```diff
@@ -118,25 +118,21 @@ func checkDiskFree(diskPath string, minFreeDisk int64) (err error) {
 	return nil
 }
 
-// removeDuplicateVols - remove duplicate volumes.
-func removeDuplicateVols(vols []VolInfo) []VolInfo {
-	length := len(vols) - 1
-	for i := 0; i < length; i++ {
-		for j := i + 1; j <= length; j++ {
-			if vols[i].Name == vols[j].Name {
-				// Pick the latest volume, if there is a duplicate.
-				if vols[i].Created.Sub(vols[j].Created) > 0 {
-					vols[i] = vols[length]
-				} else {
-					vols[j] = vols[length]
-				}
-				vols = vols[0:length]
-				length--
-				j--
-			}
+func removeDuplicateVols(volsInfo []VolInfo) []VolInfo {
+	// Use map to record duplicates as we find them.
+	encountered := make(map[string]bool)
+	result := []VolInfo{}
+
+	for i := range volsInfo {
+		if !encountered[volsInfo[i].Name] {
+			// Record this element as an encountered element.
+			encountered[volsInfo[i].Name] = true
+			// Append to result slice.
+			result = append(result, volsInfo[i])
 		}
 	}
-	return vols
+	// Return the new slice.
+	return result
 }
```

Comment:
1. If you like two for loops, you can do this by
```go
m := make(map[string]VolInfo)
for _, v := range volsInfo {
    if _, found := m[v.Name]; !found {
        m[v.Name] = v
    }
}

result := make([]VolInfo, 0, len(m))
for _, v := range m {
    result = append(result, v)
}
```

2. If you like to keep the last duplicate into result than the first one
```go
m := make(map[string]VolInfo)
for _, v := range volsInfo {
    m[v.Name] = v
}

result := make([]VolInfo, 0, len(m))
for _, v := range m {
    result = append(result, v)
}
```

unfortunately there is no function to get map's value without doing for loop